asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -197,31 +199,35 @@ def _dock(
         ],
         output_dir: Optional[Union[str, Path]] = None,
         error="skip",
+        return_for_disk_backend=False,
     ) -> list[DockingResult]:
         """
         Docking workflow using OEPosit
         """
+        if output_dir is None and return_for_disk_backend:
+            raise ValueError(
+                "Cannot specify return_for_disk_backend and not output_dir"
+            )
 
         docking_results = []
 
         for set in inputs:
+            if output_dir is not None:
+                docked_result_json_path = Path(
+                    Path(output_dir) / set.unique_name() / "docking_result.json"
+                )
+
             if (
                 set.is_cacheable
                 and (output_dir is not None)
-                and (
-                    Path(
-                        Path(output_dir) / set.unique_name() / "docking_result.json"
-                    ).exists()
-                )
+                and (docked_result_json_path.exists())
             ):
-                print(
+                logger.info(
                     f"Docking result for {set.unique_name()} already exists, reading from disk"
                 )
```

Comment:
this broke this test: https://github.com/choderalab/asapdiscovery/blob/49310f8e369c928ef416b29bcbe75522919a86e4/asapdiscovery-docking/asapdiscovery/docking/tests/test_docking_v2.py#L71-L72

because it was relying on the `out` print statement. This needs to be changed for the test to pass!

Reply:
Should be fixed now.